synapse/http/proxyagent.py

```diff
@@ -288,25 +311,35 @@ def parse_username_password(proxy: bytes) -> Tuple[Optional[ProxyCredentials], b
     return None, proxy
 
 
-def parse_host_port(hostport: bytes, default_port: int = None) -> Tuple[bytes, int]:
+def parse_proxy(
+    proxy: bytes, default_scheme: bytes = b"http", default_port: int = 1080
+) -> Tuple[bytes, bytes, int]:
     """
-    Parse the hostname and port from a proxy connection byte string.
+    Parse the scheme, hostname and port from a proxy connection byte string.
 
     Args:
-        hostport: The proxy connection string. Must be in the form 'host[:port]'.
-        default_port: The default port to return if one is not found in `hostport`.
+        proxy: The proxy connection string. Must be in the form '[scheme://]host[:port]'.
+        default_scheme: The default scheme to return if one is not found in `proxy`. Defaults to http
+        default_port: The default port to return if one is not found in `proxy`. Defaults to 1080
 
     Returns:
-        A tuple containing the hostname and port. Uses `default_port` if one was not found.
+        A tuple containing the scheme, hostname and port.
     """
-    if b":" in hostport:
-        host, port = hostport.rsplit(b":", 1)
+    # First check if we have a scheme present
+    if b"://" in proxy:
+        scheme, host = proxy.split(b"://", 1)
```

Comment:
I have thought about the problem with Python 3.9 and `urlparse`. See: https://github.com/matrix-org/synapse/pull/9119#discussion_r592441358

I would suggest to work with urlparse after all. For compatibility reasons we can check if there is a sheme and if not we add this. Somthing like this

```suggestion
    if not b"://" in proxy:
        proxy = b"".join([default_scheme, b"://", proxy])
```